app/src/androidTest/java/org/mozilla/fenix/helpers/MockLocationUpdatesRule.kt

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.helpers
+
+import android.content.Context
+import android.location.Location
+import android.location.LocationManager
+import android.os.Build
+import android.os.SystemClock
+import android.util.Log
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.platform.app.InstrumentationRegistry.getInstrumentation
+import org.junit.rules.ExternalResource
+import java.util.Date
+import kotlin.random.Random
+import org.mozilla.fenix.helpers.TestHelper.executeShellCommandBlocking
+
+private const val mockProviderName = LocationManager.GPS_PROVIDER
+
+/**
+ * Rule that sets up a mock location provider that can inject location samples
+ * straight to the device that the test is running on.
+ *
+ * Credit to the mapbox team
+ * https://github.com/mapbox/mapbox-navigation-android/blob/87fab7ea1152b29533ee121eaf6c05bc202adf02/libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/MockLocationUpdatesRule.kt
+ *
+ */
+class MockLocationUpdatesRule : ExternalResource() {
+    private val instrumentation = getInstrumentation()
```

Comment:
Let's reuse our singleton https://github.com/mozilla-mobile/fenix/blob/main/app/src/androidTest/java/org/mozilla/fenix/helpers/TestHelper.kt#L62

Reply:
I'm not sure if it would be the same thing. mDevice is a UIDevice object, and we need an Instrumentation object here to access uiAutomation which I understand is more powerful in accessing APIs: "The APIs exposed via the returned UiAutomation work across application boundaries while the APIs exposed by the instrumentation do not."

Reply:
I used what you suggested, it worked well so far, on Firebase too.